app/controllers/api_controller/parser/request.rb

```diff
@@ -0,0 +1,81 @@
+class ApiController
+  module Parser
+    class Request
+      def initialize(req, params)
+        @request = req
+        @params = params
+      end
+
+      def action
+        # for basic HTTP POST, default action is "create" with data being the POST body
+        @action ||= method == :put ? 'edit' : (json_body['action'] || 'create')
+      end
+
+      def api_prefix
+        @api_prefix ||= "#{base}#{prefix}"
+      end
+
+      def base
+        url.partition(fullpath)[0] # http://target
+      end
+
+      def collection
+        @params[:collection]
+      end
+
+      def c_id
```

Comment:
Since we're updating any code related to this here, WDYT about making this more descriptive such as `collection_id`, and similarly with `s_id` below?